### ISSUE TITLE
infinite loop (!) when installing with Roxygen2 6.1.0

### ISSUE THREAD

**Issue by infotroph** [ebimodeling/biocro#9]
with Roxygen 6.0.1:

```
0> R_LIBS_USER=~/R/library Rscript -e 'packageVersion("roxygen2"); devtools::install_github("ebimodeling/biocro")'
[1] ‘6.0.1’
Downloading GitHub repo ebimodeling/biocro@master
from URL https://api.github.com/repos/ebimodeling/biocro/zipball/master
Installing BioCro
'/Library/Frameworks/R.framework/Resources/bin/R' --no-site-file --no-environ  \
  --no-save --no-restore --quiet CMD INSTALL  \
  '/private/var/folders/qr/mbw8xxpd45jdv_46b27914280000gn/T/RtmpROx22k/devtools116bebd4476d/ebimodeling-biocro-18b9ade'  \
  --library='/Users/chrisb/R/library' --install-tests 

* installing *source* package ‘BioCro’ ...
> roxygen2::roxygenize('.', roclets='rd')
First time using roxygen2. Upgrading automatically...
Loading required package: lattice
Warning: @export [BioCro.R#199]: unknown tag
[...snip...]
Warning: @S3method [willowGro.R#728]: unknown tag
Writing BioGro.Rd
[snip]
Writing weach.Rd
Skipping invalid path:  .datatable.aware.Rd 
Writing willowGro.Rd
Writing plot.willowGro.Rd
> 
> 
** libs
clang -I/Library/Frameworks/R.framework/Resources/include -DNDEBUG   -I/usr/local/include   -fPIC  -Wall -g -O2  -c AuxBioCro.c -o AuxBioCro.o
[...snip...]
clang -dynamiclib -Wl,-headerpad_max_install_names -undefined dynamic_lookup -single_module -multiply_defined suppress -L/Library/Frameworks/R.framework/Resources/lib -L/usr/local/lib -o BioCro.so AuxBioCro.o AuxCropGro.o AuxMaizeGro.o AuxcaneGro.o AuxwillowGro.o BioCro.o CanAC.o Century.o Filling_BioCro_SoilStructure.o Functions.o RWrapper.o R_CanA.o R_MisGro.o R_SABioGro.o R_c3CanA.o R_c3photo.o R_c4photo.o R_caneGro.o R_cntry.o R_maizeGro.o R_willowGro.o assignManagement.o c3CanAC.o c3EvapoTrans.o c3photo.o c4photo.o dailywillow.o eC4photo.o leaf_nitrogen_limitation_functions.o soyGro.o -F/Library/Frameworks/R.framework/.. -framework R -Wl,-framework -Wl,CoreFoundation
installing to /Users/chrisb/R/library/BioCro/libs
** R
** data
** inst
** tests
** preparing package for lazy loading
** help
*** installing help indices
** building package indices
** installing vignettes
   ‘biocro-manual-knitr.Rnw’ 
** testing if installed package can be loaded
* DONE (BioCro)
```

With 6.1.0:

```
0> Rscript -e 'packageVersion("roxygen2"); devtools::install_github("ebimodeling/biocro")'
[1] ‘6.1.0’
Downloading GitHub repo ebimodeling/biocro@master
from URL https://api.github.com/repos/ebimodeling/biocro/zipball/master
Installing BioCro
'/Library/Frameworks/R.framework/Resources/bin/R' --no-site-file --no-environ  \
  --no-save --no-restore --quiet CMD INSTALL  \
  '/private/var/folders/qr/mbw8xxpd45jdv_46b27914280000gn/T/RtmpEqdrRA/devtools1180426cf3917/ebimodeling-biocro-18b9ade'  \
  --library='/Library/Frameworks/R.framework/Versions/3.4/Resources/library'  \
  --install-tests 

* installing *source* package ‘BioCro’ ...
> roxygen2::roxygenize('.', roclets='rd')
First time using roxygen2. Upgrading automatically...
Updating roxygen version in /private/var/folders/qr/mbw8xxpd45jdv_46b27914280000gn/T/RtmpEqdrRA/devtools1180426cf3917/ebimodeling-biocro-18b9ade/DESCRIPTION
Warning: @export [/private/var/folders/qr/mbw8xxpd45jdv_46b27914280000gn/T/RtmpEqdrRA/devtools1180426cf3917/ebimodeling-biocro-18b9ade/R/BioCro.R#199]: unknown tag
[...snip...][/private/var/folders/qr/mbw8xxpd45jdv_46b27914280000gn/T/RtmpEqdrRA/devtools1180426cf3917/ebimodeling-biocro-18b9ade/R/willowGro.R#728]: unknown tag
Loading BioCro
Re-compiling BioCro
Running /Library/Frameworks/R.framework/Resources/bin/R CMD INSTALL \
  /private/var/folders/qr/mbw8xxpd45jdv_46b27914280000gn/T/RtmpEqdrRA/devtools1180426cf3917/ebimodeling-biocro-18b9ade \
  '--library=/var/folders/qr/mbw8xxpd45jdv_46b27914280000gn/T//RtmphRxTfQ/devtools_install_1181674adad5a' \
  --no-R --no-data --no-help --no-demo --no-inst --no-docs --no-exec \
  --no-multiarch --no-test-load --preclean
* installing *source* package ‘BioCro’ ...
> roxygen2::roxygenize('.', roclets='rd')
First time using roxygen2. Upgrading automatically...
Warning: @export [/private/var/folders/qr/mbw8xxpd45jdv_46b27914280000gn/T/RtmpEqdrRA/devtools1180426cf3917/ebimodeling-biocro-18b9ade/R/BioCro.R#199]: unknown tag
[snip]
Warning: @S3method [/private/var/folders/qr/mbw8xxpd45jdv_46b27914280000gn/T/RtmpEqdrRA/devtools1180426cf3917/ebimodeling-biocro-18b9ade/R/willowGro.R#728]: unknown tag
Loading BioCro
Re-compiling BioCro
Running /Library/Frameworks/R.framework/Resources/bin/R CMD INSTALL \
  /private/var/folders/qr/mbw8xxpd45jdv_46b27914280000gn/T/RtmpEqdrRA/devtools1180426cf3917/ebimodeling-biocro-18b9ade \
  '--library=/var/folders/qr/mbw8xxpd45jdv_46b27914280000gn/T//RtmpJ8uED8/devtools_install_1183c65a96a46' \
  --no-R --no-data --no-help --no-demo --no-inst --no-docs --no-exec \
  --no-multiarch --no-test-load --preclean
* installing *source* package ‘BioCro’ ...
> roxygen2::roxygenize('.', roclets='rd')
First time using roxygen2. Upgrading automatically...
Warning: @export [/private/var/folders/qr/mbw8xxpd45jdv_46b27914280000gn/T/RtmpEqdrRA/devtools1180426cf3917/ebimodeling-biocro-18b9ade/R/BioCro.R#199]: unknown tag
[...and so on, repeating until I interrupt...]
^CLoading BioCro
Error in run(bin, args = real_cmdargs, stdout_line_callback = real_callback(stdout),  : 
  System command interrupted
Calls: <Anonymous> ... tryCatch -> tryCatchList -> tryCatchOne -> <Anonymous>
In addition: Warning message:
roxygen2 requires Encoding: UTF-8 
Execution halted
ERROR: configuration failed for package ‘BioCro’
* removing ‘/Library/Frameworks/R.framework/Versions/3.4/Resources/library/BioCro’
* restoring previous ‘/Library/Frameworks/R.framework/Versions/3.4/Resources/library/BioCro’
Installation failed: Command failed (1)
```

**Comment by infotroph** [comment]
The underlying issue here seems to be with the `roxygenize` in configure -- in Roxygen 6.1.0, it seems`roxygenize` now calls `./configure` somewhere inside itself, giving us an infinite recursion. 

One fix that looks easy from here: Remove the configure script entirely, then check in all the compiled Rd files so that users need not run Roxygen during installation at all. But perhaps there are reasons it wasn't set up that way already?

**Comment by infotroph** [comment]
Here's a hacky workaround that avoids changing any package code: Detect recursive `configure` calls and avoid re-running roxygenize. Not yet convinced it's a good idea, and have not tested on Windows. 

@justinmcgrath do you want this as a PR, and if so just here or to the dev version as well?

```
set -e

function cleanup () {
  rm  .config.lock
}
trap cleanup EXIT SIGINT SIGTERM


if [[ ! -e .config.lock ]]; then # avoid re-rentry if roxygenize calls ./configure
	touch .config.lock
	R --silent --vanilla -e "roxygen2::roxygenize('.', roclets='rd')"
fi
```

**Comment by justinmcgrath** [comment]
Generated files are not committed to the repository because then they can get out of sync with the source, so I don't want to commit the generated manuals. Does your solution work on Windows?

**Comment by infotroph** [comment]
I haven't tried yet, but I now doubt my approach is tenable. It breaks with` devtools::install_github`, which is the de facto standard installation method for R packages on GitHub these days:

```
devtools::install_github("infotroph/BioCro", ref="avoid-roxygen-loop-new-framework")
[compilation output snipped]
* DONE (BioCro)
Writing CanA.Rd
rm: .config.lock: No such file or directory
ERROR: configuration failed for package ‘BioCro’
* removing ‘/Library/Frameworks/R.framework/Versions/3.4/Resources/library/BioCro’
* restoring previous ‘/Library/Frameworks/R.framework/Versions/3.4/Resources/library/BioCro’
Writing Gro.Rd
Writing aci.Rd
Writing annualDB.Rd
Writing aq.Rd
Writing cmi0506.Rd
Writing doy124.Rd
Writing EngWea94i.Rd
Writing EngWea94rf.Rd
Writing LayET.Rd
Writing obsBeaC.Rd
Writing obsBea.Rd
Writing obsNaid.Rd
Writing simDat2.Rd
Writing WD1979.Rd
Writing weather05.Rd
Writing weather06.Rd
Writing c3photo.Rd
Writing c4photo.Rd
Writing idbp.Rd
Writing idbpm.Rd
Writing valid_dbp.Rd
> 
> 
Error in run(bin, args = real_cmdargs, stdout_line_callback = real_callback(stdout),  : 
  System command error
```

**Comment by justinmcgrath** [comment]
I suppose that since a package shouldn't be calling a different package's `configure` script, this could be submitted as a bug to roxygen, if in the meantime you can roll back to a previous version. You could also remove the configure script locally, run roxygen, then install BioCro, then restore configure.

**Comment by dlebauer** [comment]
@justinmcgrath what is the rationale for having these configure scripts? This is non-standard practice for R packages. Is there something that the standard build tools can't handle?

**Comment by justinmcgrath** [comment]
configure scripts are part of the standard R build process, and the build process is what calls the script. The script is needed because of roxygen, and the current broken state is also a problem with roxygen that should be fixed there.

**Comment by dlebauer** [comment]
@justinmcgrath after removing the configure files in PR #10, the broken build appears to be due to failing tests (e.g. https://travis-ci.org/ebimodeling/biocro/builds/442322135?utm_source=github_status&utm_medium=notification), and the tests that are broken are really tests that the model is sensitive to key physiological and soil parameters. 

Since the configure scripts in this repository only call roxygen, and the recent version of roxygen makes the configure script obsolete, I don't understand the need for the configure files. I feel like our discussion is also recursive and the only way out is to delete the no-longer needed files.

**Comment by justinmcgrath** [comment]
Are the documentation built during R CMD INSTALL with the new version of roxygen when the configure script is not present?

**Comment by dlebauer** [comment]
@justinmcgrath The standard workflow I've seen is to build the documentation and then check the generated files into the repository. 

In any case, the bug does not occur when roxygen is called from devtools::document, so I made this change in #12.

**Comment by infotroph** [comment]
I've taken the liberty of working up a competing PR that uses the standard Roxygen workflow (run before committing, commit built products). It may need some tweaking (especially around the .Call interfaces, where my experience is thin), but it should take care of the most tedious parts anyway.

**Comment by dlebauer** [comment]
Fixed by #13
